rye/src/cli/toolchain.rs

```diff
@@ -111,17 +109,6 @@ fn register(cmd: RegisterCommand) -> Result<(), Error> {
 
 /// Checks if a toolchain is still in use.
 fn check_in_use(ver: &PythonVersion) -> Result<(), Error> {
-    // Check if used by rye itself.
-    let app_dir = get_app_dir();
-    for venv in &[app_dir.join("self"), get_pip_tools_venv_path(ver)] {
```

Comment:
I don't think this whole loop should be removed, only the pip tools element in the array?

Reply:
Great catch @bluss , added this back in and trimmed down the scope of the check. https://github.com/astral-sh/rye/pull/1226/commits/139c3d88d81a99805e327c957ab5ff934918386e